packages/block-library/src/template-part/variations.js

```diff
@@ -51,10 +26,21 @@ const unsubscribe = subscribe( () => {
 			);
 			return entity?.area === variationAttributes.area;
 		};
-	} );
 
-	dispatch( blocksStore ).addBlockVariations(
-		'core/template-part',
-		variations
-	);
-} );
+		const variations = settings.variations.map( ( variation ) => {
+			return {
+				...variation,
+				...( ! variation.isActive && { isActive } ),
+				...( typeof variation.icon === 'string' && {
```

Comment:
Should we check `icons[ variation.icon ] || set the TP default icon`? The `typeof === 'string'` can be omitted. 

Also related to that, should we just create a mapper object that would contain the `header` and `footer` icons to avoid importing all the icons and would default to `layout`? --cc @gziolo

Reply:
That all makes sense.  

> typeof === 'string' can be omitted.

I was wondering if maybe there was a way someone could have registered it with an actual icon instead of an icon name, and to just double check if it was a simple string at this point before trying to overwrite it.

Reply:
>  should we just create a mapper object that would contain the header and footer icons to avoid importing all the icons and would default to layout?

Yeah, I'l go ahead and limit this to just importing the icons we expect for template parts at this time (header, footer, sidebar, and layout).

It would be neat if in the future we could enable registering custom icons with a template part area.  Its no requirement at this point, but if instead of defining `'icon' => 'header'` on the php side we could define `'icon' => $some_icon`.  But im drawing a blank on how one might go about that.

Reply:
I was planning on exploring allowing icons to be set with svg strings, but the tricky part is finding a solution that will also work with react native

Reply:
@gwwar that would be neat!  Im not too familiar with native's limitations there.